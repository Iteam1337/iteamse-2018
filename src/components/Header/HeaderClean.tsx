import React from 'react'
import { Link } from 'react-router-dom'
import styled from '../../theme'
import { GridColumnClean } from '../Grid/GridColumn'
import logo from './img/iteam_inverted.png'

const LogoLink = styled(Link)`
  &:focus {
    outline: none;
  }
`

const Wrap = styled(GridColumnClean)`
  grid-column: 2;
  grid-row: 1;
`

const Header = styled.div`
  display: grid;
  grid-row: 1;
  width: 150px;
  @media (min-width: 481px) {
    padding: 0;
  }
`

const Logo = styled.img`
  padding: 35px 0;
  max-width: 75%;
  @media (min-width: 1025px) {
    max-width: 100%;
  }
`

const Content = styled.div`
  display: grid;
  grid-column: -1 / 1;
  @media (min-width: 1025px) {
    grid-template-columns: 1fr 1024px 1fr;
  }
`

const HeaderClean = () => {
  return (
    <Wrap>
      <Content>
        <Header>
          <LogoLink to="/">
            <Logo alt="Logo" src={logo} />
          </LogoLink>
        </Header>
      </Content>
    </Wrap>
  )
}

export default HeaderClean
