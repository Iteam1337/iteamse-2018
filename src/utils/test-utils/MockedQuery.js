import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import ApolloClient from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

type P = {
  children: React.Node,
  loading?: boolean,
  response: Object,
}

const MockedQuery = ({ children, loading, response = {} }: P) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    link: createHttpLink({
      uri: 'http://localhost:3000',
      fetch: mockedNetworkFetch,
    }),
    cache: new InMemoryCache(),
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
