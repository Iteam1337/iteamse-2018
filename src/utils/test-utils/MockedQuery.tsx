import { InMemoryCache } from 'apollo-cache-inmemory'
import ApolloClient from 'apollo-client'
import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { mockSingleLink } from 'react-apollo/test-utils'
import MockRouter from 'react-mock-router'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

interface MockedQueryProps {
  location?: object
  mocks: any[]
  params?: object
}

const MockedQuery: React.SFC<MockedQueryProps> = ({
  children,
  location,
  mocks,
  params,
}) => {
  const link = mockSingleLink.apply(null, mocks)

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <MockRouter location={location} params={params}>
          {children}
        </MockRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
