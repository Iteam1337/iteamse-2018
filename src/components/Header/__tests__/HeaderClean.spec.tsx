import * as React from 'react'
import { render, wait } from 'react-testing-library'
import MockedQuery from '../../../utils/test-utils/MockedQuery'
import HeaderClean from '../HeaderClean'

describe('components/HeaderClean', () => {
  const mocks = {}

  it('renders HeaderClean', async () => {
    const { container } = render(
      <MockedQuery mocks={mocks}>
        <HeaderClean />
      </MockedQuery>
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
