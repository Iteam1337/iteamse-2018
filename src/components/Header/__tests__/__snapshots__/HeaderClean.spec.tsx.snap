// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HeaderClean renders HeaderClean 1`] = `
<div>
  <div
    class="GridColumn__GridColumnClean-sc-1xz61ww-0 sc-htpNat fLNxzH"
  >
    <div
      class="HeaderClean__Content-sc-135p41j-2 ijPUvR"
    >
      <div
        class="HeaderClean__Header-sc-135p41j-0 iowroM"
      >
        <a
          class="sc-bwzfXH crrApc"
        >
          <img
            alt="Logo"
            src="test-file-stub"
            class="HeaderClean__Logo-sc-135p41j-1 iIOSxZ"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
